src/util/simplify_expr.cpp

```diff
@@ -2423,30 +2420,24 @@ simplify_exprt::simplify_byte_update(const byte_update_exprt &expr)
   return unchanged(expr);
 }
 
-bool simplify_exprt::simplify_complex(exprt &expr)
+simplify_exprt::resultt<> simplify_exprt::simplify_complex(const exprt &expr)
 {
   if(expr.id() == ID_complex_real)
```

Comment:
Shouldn't this be two separate functions rather than branching twice on the same id in quick succession? (Once to figure out we need to call `simplify_complex` and then this one here. We could then have a typesafe interface.)

Reply:
Sure, can be split; will add to #4874